iothub/device/src/Common/SynchronizedPool.cs

```diff
@@ -320,45 +310,42 @@ T TakeFromPerThreadPool(int thisThreadID)
             return null;
         }
 
-        T TakeFromGlobalPool(int thisThreadID)
+        private T TakeFromGlobalPool(int thisThreadID)
         {
             RecordTakeFromGlobalPool(thisThreadID);
 
             return globalPool.Take();
         }
 
-        struct Entry
+        private struct Entry
         {
             public int threadID;
             public T value;
         }
 
-        struct PendingEntry
+        private struct PendingEntry
         {
             public int returnCount;
             public int threadID;
         }
 
-        static class SynchronizedPoolHelper
+        private static class SynchronizedPoolHelper
         {
             public static readonly int ProcessorCount = GetProcessorCount();
 
             [Fx.Tag.SecurityNote(Critical = "Asserts in order to get the processor count from the environment", Safe = "This data isn't actually protected so it's ok to leak")]
-#if NET451
-            [EnvironmentPermission(SecurityAction.Assert, Read = "NUMBER_OF_PROCESSORS")]
```

Comment:
Shouldn't we keep this line? We aren't removing support for net451 yet, right?

Reply:
thank you for finding this. I don't know how they get removed. It has happened on me several times.